Maybe this is caused by JVM-3 not responding. I had assumed that this was JVM-4 shutting down after leave due to the following logs:

```

[JVM-4] [INFO] [10/30/2017 08:24:55.999] [MultiDcSpec-akka.actor.default-dispatcher-16] [akka.cluster.Cluster(akka://MultiDcSpec)] Cluster Node [akka.trttl.gremlin.tcp://MultiDcSpec@a5.moxie:44224] dc [dc2] - Exiting (leader), starting coordinated shutdown
[JVM-4] [INFO] [10/30/2017 08:24:56.001] [MultiDcSpec-akka.actor.default-dispatcher-16] [akka.cluster.Cluster(akka://MultiDcSpec)] Cluster Node [akka.trttl.gremlin.tcp://MultiDcSpec@a5.moxie:44224] dc [dc2] - Leader is moving node [akka.trttl.gremlin.tcp://MultiDcSpec@a5.moxie:44224] to [Exiting]
[JVM-4] [INFO] [10/30/2017 08:24:56.019] [MultiDcSpec-akka.actor.default-dispatcher-3] [akka.cluster.Cluster(akka://MultiDcSpec)] Cluster Node [akka.trttl.gremlin.tcp://MultiDcSpec@a5.moxie:44224] dc [dc2] - Exiting completed
[JVM-4] [INFO] [10/30/2017 08:24:56.025] [MultiDcSpec-akka.actor.default-dispatcher-3] [akka.cluster.Cluster(akka://MultiDcSpec)] Cluster Node [akka.trttl.gremlin.tcp://MultiDcSpec@a5.moxie:44224] dc [dc2] - Shutting down...
[JVM-4] [INFO] [10/30/2017 08:24:56.028] [MultiDcSpec-akka.actor.default-dispatcher-3] [akka.cluster.Cluster(akka://MultiDcSpec)] Cluster Node [akka.trttl.gremlin.tcp://MultiDcSpec@a5.moxie:44224] dc [dc2] - Successfully shut down
[JVM-4] [INFO] [10/30/2017 08:24:56.031] [MultiDcSpec-akka.actor.default-dispatcher-3] [akka.cluster.Cluster(akka://MultiDcSpec)] Cluster Node [akka.trttl.gremlin.tcp://MultiDcSpec@a5.moxie:44224] dc [dc2] - Exiting, starting coordinated shutdown
[JVM-4] [WARN] [10/30/2017 08:24:56.056] [MultiDcSpec-akka.actor.default-dispatcher-4] [akka://MultiDcSpec/system/cluster/core/publisher] received dead letter from Actor[akka://MultiDcSpec/system/cluster/core/daemon#1180488646]: PublishChanges(MembershipState(Gossip(members = [Member(address = akka.trttl.gremlin.tcp://MultiDcSpec@a0.moxie:57019, dataCenter = dc1, status = Up), Member(address = akka.trttl.gremlin.tcp://MultiDcSpec@a5.moxie:44224, dataCenter = dc2, status = Exiting), Member(address = akka.trttl.gremlin.tcp://MultiDcSpec@a5.moxie:47905, dataCenter = dc1, status = Up), Member(address = akka.trttl.gremlin.tcp://MultiDcSpec@a6.moxie:38137, dataCenter = dc2, status = Up), Member(address = akka.trttl.gremlin.tcp://MultiDcSpec@a6.moxie:58545, dataCenter = dc2, status = Up)], overview = GossipOverview(reachability = [], seen = [UniqueAddress(akka.trttl.gremlin.tcp://MultiDcSpec@a6.moxie:58545,278800600), UniqueAddress(akka.trttl.gremlin.tcp://MultiDcSpec@a6.moxie:38137,-1298477567), UniqueAddress(akka.trttl.gremlin.tcp://MultiDcSpec@a5.moxie:44224,588429241)]), version = VectorClock(27d37f47e6bd5687562efc023a9a1cec -> 3, 42b8582297ab98331047fcc83b02dce6 -> 3, 48f4f0fd8829a1723ad92f52d7389f92 -> 6), tombstones = Map()),UniqueAddress(akka.trttl.gremlin.tcp://MultiDcSpec@a5.moxie:44224,588429241),dc2,5))
[JVM-4] [INFO] [10/30/2017 08:24:56.065] [MultiDcSpec-akka.remote.default-remote-dispatcher-6] [akka.trttl.gremlin.tcp://MultiDcSpec@a5.moxie:44224/system/remoting-terminator] Shutting down remote daemon.
[JVM-4] [INFO] [10/30/2017 08:24:56.068] [MultiDcSpec-akka.remote.default-remote-dispatcher-6] [akka.trttl.gremlin.tcp://MultiDcSpec@a5.moxie:44224/system/remoting-terminator] Remote daemon shut down; proceeding with flushing remote transports.
[JVM-1] [ERROR] [10/30/2017 08:24:56.077] [MultiDcSpec-akka.actor.default-dispatcher-18] [akka://MultiDcSpec/user/controller/barriers] unannounced disconnect of RoleName(fourth) (akka.remote.testconductor.BarrierCoordinator$ClientLost: unannounced disconnect of RoleName(fourth))
```

But JVM-3 doesn't appear to do anything for 3s.

Anyways i've turned on DEBUG logging in the mean time

--- a/akka-cluster/src/multi-jvm/scala/akka/cluster/MultiDcClusterSpec.scala
+++ b/akka-cluster/src/multi-jvm/scala/akka/cluster/MultiDcClusterSpec.scala
@@ -19,7 +19,9 @@ class MultiDcSpecConfig(crossDcConnections: Int = 5) extends MultiNodeConfig {
 
   commonConfig(ConfigFactory.parseString(
     s"""
-      akka.loglevel = INFO
+      # DEBUG On for issue #23864
+      akka.loglevel = DEBUG
+      akka.coordinated-shutdown.terminate-actor-system = off
       akka.cluster.multi-data-center.cross-data-center-connections = $crossDcConnections
     """).withFallback(MultiNodeClusterSpec.clusterConfig))
 
@@ -71,7 +73,7 @@ abstract class MultiDcSpec(config: MultiDcSpecConfig)
 
       runOn(first, second, third, fourth) {
         within(20.seconds) {
-          awaitAssert(clusterView.members.filter(_.status == MemberStatus.Up) should have size (4))
+          awaitAssert(clusterView.members.filter(_.status == MemberStatus.Up) should have size 4)
         }
       }
 
@@ -85,6 +87,7 @@ abstract class MultiDcSpec(config: MultiDcSpecConfig)
         val dc1 = Set(address(first), address(second))
         dc1 should contain(clusterView.leader.get)
       }
+
       runOn(third, fourth) {
         cluster.settings.SelfDataCenter should ===("dc2")
         clusterView.leader shouldBe defined
@@ -109,7 +112,7 @@ abstract class MultiDcSpec(config: MultiDcSpecConfig)
         // should be able to join and become up since the
         // unreachable is between dc1 and dc2,
         within(10.seconds) {
-          awaitAssert(clusterView.members.filter(_.status == MemberStatus.Up) should have size (5))
+          awaitAssert(clusterView.members.filter(_.status == MemberStatus.Up) should have size 5)
         }
       }
 
@@ -120,7 +123,7 @@ abstract class MultiDcSpec(config: MultiDcSpecConfig)
       // should be able to join and become up since the
       // unreachable is between dc1 and dc2,
       within(10.seconds) {
-        awaitAssert(clusterView.members.filter(_.status == MemberStatus.Up) should have size (5))
+        awaitAssert(clusterView.members.filter(_.status == MemberStatus.Up) should have size 5)
       }
 
       enterBarrier("inter-data-center unreachability end")
@@ -133,12 +136,13 @@ abstract class MultiDcSpec(config: MultiDcSpecConfig)
       enterBarrier("other-data-center-internal-unreachable")
 
       runOn(third) {
+        // FIXME This is already part of the cluster, is this intended? Joined on line 107
         cluster.join(fifth)
         // should be able to join and leave
         // since the unreachable nodes are inside of dc1
         cluster.leave(fourth)
 
-        awaitAssert(clusterView.members.map(_.address) should not contain (address(fourth)))
+        awaitAssert(clusterView.members.map(_.address) should not contain address(fourth))
         awaitAssert(clusterView.members.collect { case m if m.status == Up ⇒ m.address } should contain(address(fifth)))
       }
 
@@ -147,7 +151,7 @@ abstract class MultiDcSpec(config: MultiDcSpecConfig)
       runOn(first) {
         testConductor.passThrough(first, second, Direction.Both).await
       }
-      enterBarrier("other-datac-enter-internal-unreachable end")
+      enterBarrier("other-data-center-internal-unreachable end")
     }
 
     "be able to down a member of another data-center" in within(20.seconds) {
@@ -156,10 +160,9 @@ abstract class MultiDcSpec(config: MultiDcSpecConfig)
       }
 
       runOn(first, third, fifth) {
-        awaitAssert(clusterView.members.map(_.address) should not contain (address(second)))
+        awaitAssert(clusterView.members.map(_.address) should not contain address(second))
       }
       enterBarrier("cross-data-center-downed")
     }
-
   }
 }

--- a/akka-cluster/src/multi-jvm/scala/akka/cluster/NodeLeavingAndExitingSpec.scala
+++ b/akka-cluster/src/multi-jvm/scala/akka/cluster/NodeLeavingAndExitingSpec.scala
@@ -3,12 +3,9 @@
  */
 package akka.cluster
 
-import scala.collection.immutable.SortedSet
-import com.typesafe.config.ConfigFactory
 import akka.remote.testkit.MultiNodeConfig
 import akka.remote.testkit.MultiNodeSpec
 import akka.testkit._
-import scala.concurrent.duration._
 import akka.actor.Props
 import akka.actor.Actor
 import akka.cluster.MemberStatus._
@@ -62,7 +59,6 @@ abstract class NodeLeavingAndExitingSpec
 
         // Verify that 'second' node is set to EXITING
         exitingLatch.await
-
       }
 
       // node that is leaving

